drivers/lock.py

```diff
@@ -163,20 +213,29 @@ def acquire(self):
         """Blocking lock aquisition, with warnings. We don't expect to lock a
         lot. If so, not to collide. Coarse log statements should be ok
         and aid debugging."""
-        if not self.lock.trylock():
-            util.SMlog("Failed to lock %s on first attempt, " % self.lockpath
-                   + "blocked by PID %d" % self.lock.test())
-            self.lock.lock()
-        if VERBOSE:
-            util.SMlog("lock: acquired %s" % self.lockpath)
+        if not self.held():
+            if not self.lock.trylock():
+                util.SMlog("Failed to lock %s on first attempt, " % self.lockpath
+                       + "blocked by PID %d" % self.lock.test())
+                self.lock.lock()
+            if VERBOSE:
+                util.SMlog("lock: acquired %s" % self.lockpath)
+        self.count += 1
 
     def acquireNoblock(self):
         """Acquire lock if possible, or return false if lock already held"""
-        exists = os.path.exists(self.lockpath)
-        ret = self.lock.trylock()
-        if VERBOSE:
-            util.SMlog("lock: tried lock %s, acquired: %s (exists: %s)" % \
-                    (self.lockpath, ret, exists))
+        if not self.held():
+            exists = os.path.exists(self.lockpath)
```

Comment:
Move this line to bellow the 

if VERBOSE

Reply:
errr, no, that's not what we want.

Reply:
But the variable "exists" is only used in the log comments, unless i've missed something.

Reply:
but it needs to be done before taking the lock as it's guaranteed to exist after taking the lock.

Reply:
Dohh, understood